src/test/kotlin/org/rust/lang/core/resolve/RsResolveTestBase.kt

```diff
@@ -41,27 +45,33 @@ abstract class RsResolveTestBase : RsTestBase() {
         mark.checkHit { checkByCode(code) }
 
     protected fun stubOnlyResolve(@Language("Rust") code: String) {
-        val testProject = fileTreeFromText(code)
-            .createAndOpenFileWithCaretMarker()
+        stubOnlyResolve<RsWeakReferenceElement>(fileTreeFromText(code))
+    }
+
+    protected fun stubOnlyResolve(@Language("Rust") code: String, mark: Testmark) =
+        mark.checkHit { stubOnlyResolve(code) }
+
+    protected inline fun <reified T: PsiElement> stubOnlyResolve(fileTree: FileTree) {
```

Comment:
I think it's a bad practice to create large inline functions. It the case of reified types it's simple to split it into 2 methods like here https://github.com/intellij-rust/intellij-rust/pull/2647/files#diff-c2a83b6bef270a72d06a7ba0972ecf61R56